drivers/adc/adc_nrfx_adc.c

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+#include <nrfx_adc.h>
+
+#define SYS_LOG_DOMAIN "adc_nrfx_adc"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ADC_LEVEL
+#include <logging/sys_log.h>
+
+struct adc_nrfx_data {
+	struct adc_context ctx;
+	nrf_adc_value_t *buffer;
+	u8_t active_channels;
+};
+
+static struct adc_nrfx_data m_data = {
+	ADC_CONTEXT_INIT_TIMER(m_data, ctx),
+	ADC_CONTEXT_INIT_LOCK(m_data, ctx),
+	ADC_CONTEXT_INIT_SYNC(m_data, ctx),
+};
+
+static nrfx_adc_channel_t m_channels[CONFIG_ADC_NRFX_ADC_CHANNEL_COUNT];
+
+
+static int adc_nrfx_channel_setup(struct device *dev,
```

Comment:
use nrfx_adc_ as a function prefix, its a bit confusing when seeing adc_ and thinking maybe that's an adc core function, and not a driver specific one.

Reply:
But the `nrfx_` prefix is used for stuff that comes from nrfx, and in particular all API functions of the nrfx driver for ADC peripheral are prefixed with `nrfx_adc_`. So that would be confusing as well. On the other hand, `adc_nrfx_` is consistent with the scheme we use for naming implementation files.
And as I see such approach is commonly used. See for example: `spi_dw_transceive`, `i2c_mcux_configure`, `uart_stm32_poll_in`. Thus, I would prefer to stay with this naming.

Reply:
ok, that makes sense.

Reply:
I guess my comment is about being consistent in this file, so either everything should be prefixed adc_nrfx_ or not.